### DISCUSSION TITLE
Add file searching to default behavior; potential customization and power-ups down the road?

### DISCUSSION THREAD

**Discussion by brandonbeaumia** [pop-os/launcher#85] [Ideas]
Having to specify a tool/function/plugin for Launcher queries is not easy to discover and differs from most "find anything" search paradigms. As a quick stop-gap measure, I would love to see file search added to the no-keyword results. This would smooth the transition to Pop for a lot of people who are used to their search boxes returning both apps and files.

I have some additional thoughts for customization down the road. A GUI to manage plugins and customize their keywords would be great. Ideally, though, most plugins would not require a keyword. Instead, could plugins register their data types/capability with the Launcher instead? Even a regex might be enough of a filter.

Numbers and symbols only? Calculate an answer. Numbers, letters, and symbols? Send to custom web search (WolframAlpha in my case.) Text? App launcher and files and web search (for example).

Something that is rarely used or resource intensive? Keyword by default, but if you've got a screamin' NVME drive and 8c/16t to work with, a user could also switch a radio button from "Keyword: [text box]" to something like "Process all Searches". And maybe you almost never use the launcher for file search or have a old spinning drive? Then a user could add the keyword back in to prevent slowdown or clutter in the results.

Anyway, just spitballing a bit here. The S76 team seems to be full of exactly my kind of power user, and if Launcher was treated as a core feature of the desktop with resources allocated as such I could see it becoming Pop's "killer feature."

**Comment by mmstick** [comment]
It's not required for a plugin to need a keyword, and all searches are matched by a regular expression.

**Reply by brandonbeaumia** [reply]
That's great!! After some searching I did find a Reddit thread on rebinding the shortcuts; I will try to follow up tonight and get this working. The interface for plugin management and options would still be cool but if I can add 'find' to non-keyword results through a config file that would alleviate most of my frustration as I shift back and forth between Windows and Linux daily. Thank you!
